activestorage/lib/active_storage/attached/one.rb

```diff
@@ -54,16 +54,9 @@ def blank?
     #   person.avatar.attach(io: File.open("/path/to/face.jpg"), filename: "face.jpg", content_type: "image/jpeg")
     #   person.avatar.attach(avatar_blob) # ActiveStorage::Blob object
     def attach(attachable)
-      if record.persisted? && !record.changed?
-        record.public_send("#{name}=", attachable)
-        if record.save
-          record.public_send("#{name}")
-        else
-          false
-        end
-      else
-        record.public_send("#{name}=", attachable)
-      end
+      record.public_send("#{name}=", attachable)
+      return false if record.persisted? && !record.changed? && !record.save
+      record.public_send("#{name}")
```

Comment:
Sidenote: the code I wrote in https://github.com/rails/rails/pull/45345#discussion_r899472836 was to illustrate my point about consistency of the return value.  I would also be happy with something like the following if it is more clear (and it may well be):

```ruby
record.public_send("#{name}=", attachable)
if record.persisted? && !record.changed?
  return if !record.save
end
record.public_send("#{name}")
```

I can even see an argument for keeping two explicit branches, because it is an unrelated detail that `record.changed?` is still false after `record.public_send("#{name}=", ...)`:

```ruby
if record.persisted? && !record.changed?
  record.public_send("#{name}=", attachable)
  return if !record.save
else
  record.public_send("#{name}=", attachable)
end
record.public_send("#{name}")
```

Reply:
I like the version where `record.save` is separate from the predicate methods.

I understand your point for the last example, but I'm not sure if it's worth keeping that more complex version or assuming that calling the ActiveStorage setter method on an active record object won't change the record itself.. or maybe we could add a test for that? Because if someday it changes, it'll definitively break the `attach` method